test_conformance/api/test_api_min_max.cpp

```diff
@@ -1946,64 +1946,90 @@ int test_min_max_device_version(cl_device_id deviceID, cl_context context, cl_co
+        for (i = 0;
+             i < sizeof(requiredExtensions11) / sizeof(requiredExtensions11[0]);
+             i++)
         {
-            if(!is_extension_available(deviceID, requiredExtensions[i]))
+            if (!is_extension_available(deviceID, requiredExtensions11[i]))
             {
-                log_error( "ERROR: Required extension for 1.1 and greater devices is not in extension string: %s\n", requiredExtensions[i] );
+                log_error("ERROR: Required extension for 1.1 and greater "
+                          "devices is not in extension string: %s\n",
+                          requiredExtensions11[i]);
                 return -1;
             }
             else
-                log_info( "\t%s\n", requiredExtensions[i] );
+                log_info("\t%s\n", requiredExtensions11[i]);
         }
 
-        if( major >= 2 )
+        if (version_as_integer >= 12)
         {
-            log_info( "Checking for required extensions for OpenCL 2.0 and later devices...\n" );
-
-            // Check if double precision is supported, if it is, then check the extension "cl_khr_fp64"
-            cl_device_fp_config value;
-            int numRequiredExtension20 = sizeof(requiredExtensions20)/sizeof(char *);
-
-            error = clGetDeviceInfo( deviceID, CL_DEVICE_DOUBLE_FP_CONFIG, sizeof( value ), &value, 0 );
+            log_info("Checking for required extensions for OpenCL 1.2 and "
+                     "later devices...\n");
+            // The only required extension for an OpenCL-1.2 device is
+            // cl_khr_fp64 and it is only required if double precision is
+            // supported.
+            cl_device_fp_config doubles_supported;
+            error = clGetDeviceInfo(deviceID, CL_DEVICE_DOUBLE_FP_CONFIG,
+                                    sizeof(doubles_supported),
+                                    &doubles_supported, 0);
             test_error( error, "Unable to get device double fp config" );
+            if (doubles_supported)
+            {
+                if (!is_extension_available(deviceID, "cl_khr_fp64"))
+                {
+                    log_error(
+                        "ERROR: Required extension for 1.2 and greater devices "
+                        "is not in extension string: cl_khr_fp64\n");
+                }
+                else
+                {
+                    log_info("\t%s\n", "cl_khr_fp64");
+                }
+            }
+        }
 
-            // if double precision is not supported, then don't check extension "cl_khr_fp64"
-            numRequiredExtension20 -= (value == 0);
-
-            for( i = 0; i<numRequiredExtension20; i++ )
+        if (version_as_integer >= 20 && version_as_integer < 30)
+        {
+            log_info("Checking for required extensions for OpenCL 2.0, 2.1 and "
+                     "2.2 devices...\n");
+            for (i = 0; i < sizeof(requiredExtensions2x)
+                     / sizeof(requiredExtensions2x[0]);
```

Comment:
`ARRAY_SIZE`

Reply:
Thanks for your feedback, I have updated to use `ARRAY_SIZE`.